### PR TITLE
chore(picker): fix for overlaid picker in mobile

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description
`type=auto` was taking up the root dialog component as its parent and rendering the `sp-tray` considering dialog as its root instead of the DOM.

## Related issue(s)

- https://github.com/adobe/spectrum-web-components/issues/3834

## How has this been tested?

-   [ ] _Test case 1_
    Do this:
    1. Add the below code in `picker.stories.ts` and try to emulate it in a iphone simulatorwith `xcode`.
      ```
      <overlay-trigger type="modal" open="click">
        <div slot="click-content" role="dialog">
          <sp-underlay style="z-index: 10000;"></sp-underlay>
          <div style="  position: fixed;top: 50%;left: 50%;transform: translate(-50%, -50%);z-index: 10001; background-color: yellow; padding: 20px">
            <div style="display: flex; flex-direction: column; gap: 40px">
              <sp-picker id="picker-m" size="m" label="Selection type" ${spreadProps(args)}>
                <sp-menu-item>Deselect</sp-menu-item>
                <sp-menu-item>Select inverse</sp-menu-item>
                <sp-menu-item>Feather...</sp-menu-item>
                <sp-menu-item>Select and mask...</sp-menu-item>
                <sp-menu-divider></sp-menu-divider>
                <sp-menu-item>Save selection</sp-menu-item>
                <sp-menu-item disabled>Make work path</sp-menu-item>
              </sp-picker>
              <sp-picker id="picker-m" size="m" label="Selection type">
                <sp-menu-item>Deselect</sp-menu-item>
                <sp-menu-item>Select inverse</sp-menu-item>
                <sp-menu-item>Feather...</sp-menu-item>
                <sp-menu-item>Select and mask...</sp-menu-item>
                <sp-menu-item>Save selection</sp-menu-item>
                <sp-menu-item disabled>Make work path</sp-menu-item>
              </sp-picker>
            </div>
          </div>
      </overlay-trigger>
      ```

## Screenshots (if appropriate)

Before Fix:

Screenshot 2023-11-29 at 8 58 28 PM

After Fix:
Screenshot 2023-11-29 at 8 58 17 PM

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/picker/src/Picker.ts [hunk 1]
```diff
@@ -393,7 +393,7 @@ export class PickerBase extends SizedMixin(Focusable, { noDefaultSize: true }) {
                 .offset=${0}
                 ?open=${this.open}
                 .placement=${this.isMobile.matches ? undefined : this.placement}
-                type="auto"
```
Comment (Westbrook) [inline packages/picker/src/Picker.ts:396]: Can we bind this as a property if it is going to be dynamic?

Comment (Rajdeepc) [reply packages/picker/src/Picker.ts:396]: Yes sure! I can add in OverlayTypes from Overlay and create a property in Picker

**Comment by Rajdeepc** [comment]
> here’s a change to the timing of the test/element that is needed

I dont think this is failing due to change in type. I verified in a test PR. I can see the same VRT failures

**Comment by Westbrook** [comment]
Please share why you think the change I referred to is not caused by this PR.

**Comment by Rajdeepc** [comment]
> Please share why you think the change I referred to is not caused by this PR.

I dont find any visual differences on the first load of PR storybook and PROD storybook
PR: https://bug-overlay-picker--spectrum-web-components.netlify.app/storybook/?path=/story/picker--custom
PROD: https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/picker--custom

As I can see from the failed VRT it is triggering a hover state

File: packages/picker/src/Picker.ts [hunk 2]
```diff
@@ -110,6 +110,9 @@ export class PickerBase extends SizedMixin(Focusable, { noDefaultSize: true }) {
     @property()
     public placement: Placement = 'bottom-start';
 
+    @property()
+    public overlayType: OverlayTypes = 'auto';
```
[the remark this answers is not in the document]
Comment (Rajdeepc) [reply packages/picker/src/Picker.ts]: Adding as ``` .type=${this.isMobile.matches ? 'modal' : 'auto'}```

File: .circleci/config.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ executors:
 parameters:
     current_golden_images_hash:
         type: string
-        default: d8be48bd925210181c3ec6933abbc3aade090bcb
+        default: a35e91536b6df008822b7066e413b89d6a65af85
```
Comment (Westbrook) [inline .circleci/config.yml:13]: If the decorator worked, I'd have expected this not to be needed... 🤔

**Comment by Rajdeepc** [comment]
@Westbrook Mission Accomplished!!

**Review by Westbrook — APPROVED** [review]
Love it! Thanks for digging in on this one. :shipit:

**Pull request merged**